packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -990,21 +1053,39 @@ export class ContainerRuntime
 
 		const mc = loggerToMonitoringContext(logger);
 
+		// Options without overlap with IProtocolOptions
 		const {
 			summaryOptions = {},
 			gcOptions = {},
 			loadSequenceNumberVerification = "close",
 			flushMode = defaultFlushMode,
+			maxBatchSizeInBytes = defaultMaxBatchSizeInBytes,
+			chunkSizeInBytes = defaultChunkSizeInBytes,
+			protocolOptions = {},
+			compatibilityMode = defaultCompatibilityMode,
+		}: IContainerRuntimeOptionsInternal = runtimeOptions;
+		// Options with overlap with IProtocolOptions
+		let {
 			compressionOptions = runtimeOptions.enableGroupedBatching === false
 				? disabledCompressionConfig // Compression must be disabled if Grouping is disabled
 				: defaultCompressionConfig,
-			maxBatchSizeInBytes = defaultMaxBatchSizeInBytes,
 			enableRuntimeIdCompressor,
-			chunkSizeInBytes = defaultChunkSizeInBytes,
 			enableGroupedBatching = true,
 			explicitSchemaControl = false,
 		}: IContainerRuntimeOptionsInternal = runtimeOptions;
 
+		// While we transition to using IProtocolOptions, we will need to handle duplicate properties.
+		// For now, any properties defined in protocolOptions will overwrite duplicate properties in runtimeOptions.
```

Comment:
It may be better to throw an error if we get a mismatch in the overlapping properities